src/reducers/screenReducers/posts.js

```diff
@@ -0,0 +1,71 @@
+/* @flow */
+import {
+  ADD_POST,
+  DELETE_POST,
+  UPDATE_POST,
+  GET_POSTS
+} from "../../constants/actionTypes";
+import type {
+  Post,
+  AddPostAction,
+  DeletePostAction,
+  UpdatePostAction,
+  GetPostsAction
+} from "../../types/types";
+
+type State = {
+  +posts: Array<Post>
+};
+
+export type Action =
+  | AddPostAction
+  | DeletePostAction
+  | UpdatePostAction
+  | GetPostsAction;
+
+const postsState = { posts: [] };
+
+function cloneObject(object) {
+  return JSON.parse(JSON.stringify(object));
```

Comment:
Don't use it to avoid performance issues, clone objects and arrays manually where needed with spread operators, like
`{...someObject}` or `[...someArray]`

Reply:
This relates to all similar situations